ironmq/src/main/scala/akka/stream/alpakka/ironmq/IronMqClient.scala

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.ironmq
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.Http.HostConnectionPool
+import akka.http.scaladsl.client.RequestBuilding._
+import akka.http.scaladsl.model.headers.{ Authorization, GenericHttpCredentials }
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse, Uri }
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.http.scaladsl.util.FastFuture
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{ Flow, Sink, Source }
+import akka.{ Done, NotUsed }
+import com.typesafe.config.Config
+import de.heikoseeberger.akkahttpcirce.CirceSupport._
+import io.circe.Json
+import io.circe.syntax._
+import cats.syntax.either._
+
+import scala.concurrent.duration.{ Duration, FiniteDuration }
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.{ Failure, Success, Try }
+
+object IronMqClient {
+
+  def apply()(implicit actorSystem: ActorSystem): IronMqClient =
+    apply(actorSystem.settings.config)
+
+  def apply(config: Config)(implicit actorSystem: ActorSystem): IronMqClient =
+    apply(IronMqSettings(config))
+
+  def apply(settings: IronMqSettings)(implicit actorSystem: ActorSystem): IronMqClient =
+    new IronMqClient(settings)
+
+}
+
+class IronMqClient(settings: IronMqSettings)(implicit actorSystem: ActorSystem) extends AutoCloseable {
+
+  import Codec._
+
+  private implicit val materializer = ActorMaterializer()
```

Comment:
It is used for the Akka HTTP. Not sure what is the best approach in this case.  I guess as the actor system is passed in the constructor, the materializer should be passed to the constructor as well.